lib/signs/utilities/messages.ex

```diff
@@ -258,6 +267,42 @@ defmodule Signs.Utilities.Messages do
     end
   end
 
+  defp overnight_period?(current_time, config, service_ended) do
+    first_departure =
+      RealtimeSigns.headway_engine().get_first_scheduled_departure(
```

Comment:
I think we ideally want to try limiting accessing state (calling out to engines) to the main `realtime.ex`. We actually already pass in the first scheduled departure to the `get_messages` function via the `scheduled` variable (admittedly could be named a little better). We can fetch the last scheduled departure as well and pass that along too

Reply:
Moving all of the calls into realtime.ex (recent departures too). Also did a little renaming of scheduled if that's okay.